src/recursion.js

```diff
@@ -3,16 +3,31 @@
 const nFibonacci = (n) => {
   // fibonacci sequence: 1 2 3 5 8 13 ...
   // return the nth number in the sequence
+  if (n < 3) return 1;
+  return nFibonacci(n - 2) + nFibonacci(n - 1);
 };
 
 const nFactorial = (n) => {
   // factorial example: !5 = 5 * 4 * 3 * 2 * 1
   // return the factorial of `n`
+  if (n === 1) return 1;
+  return (n * nFactorial(n - 1));
 };
 
 const checkMatchingLeaves = (obj) => {
   // return true if every property on `obj` is the same
   // otherwise return false
+  const sameArray = [];
+  Object.keys(obj).forEach((element) => {
+    if (typeof element === 'object') {
+      checkMatchingLeaves(element);
+    }
+    sameArray.push(element);
+  });
+  sameArray.forEach((element) => {
+    if (element !== sameArray[0]) return false;
```

Comment:
Because you are inside of a callback function that you've passed to `forEach` your `return` statement will not cause `checkMatchingLeaves` to exit but rather the callback.  You'd need an external variable that's in `checkMatchingLeaves` scope but not defined inside of the callback you gave to the `forEach`.